pandas/core/frame.py

```diff
@@ -4922,21 +4922,30 @@ def _arith_op(left, right):
                                  copy=False)
 
     def _combine_match_index(self, other, func, level=None):
+        assert isinstance(other, Series)
         left, right = self.align(other, join='outer', axis=0, level=level,
                                  copy=False)
+        assert left.index.equals(right.index)
```

Comment:
Are there cases where this might no be true? If not, then let's remove it, as it could be somewhat expensive (though maybe not, since the output of align shares indices.)

Reply:
This always holds.  I added it largely so I wouldn't have to keep going back to core.ops to double-check what was being passed.